src/engine/Actor.ts

```diff
@@ -1,1128 +1,1126 @@
-              
-       // Update action queue
-       this.actionQueue.update(delta);
-
-       // Update color only opacity
-       if (this.color) {
-          this.color.a = this.opacity;
-       }     
-         
-       // calculate changing opacity
-       if (this.previousOpacity !== this.opacity) {
-          this.previousOpacity = this.opacity;
-          this._opacityFx.opacity = this.opacity;          
-          this._effectsDirty = true;
-       }
-
-       // Capture old values before integration step updates them
-       this.oldVel.setTo(this.vel.x, this.vel.y);
-       this.oldPos.setTo(this.pos.x, this.pos.y);
-
-       // Run Euler integration
-       this.integrate(delta);
-
-       // Update actor pipeline (movement, collision detection, event propagation, offscreen culling)
-       for (var trait of this.traits) {
-          trait.update(this, engine, delta);
-       }
-
-       // Update child actors
-       for (var i = 0; i < this.children.length; i++) {   
-          if (this.children[i] instanceof UIActor) {
-             engine.stats.currFrame.actors.ui++;
-          } else {
-             engine.stats.currFrame.actors.alive++;
-          }
-          this.children[i].update(engine, delta);          
-       }
-       
-       // TODO: Obsolete `update` event on Actor
-       this.eventDispatcher.emit('update', new PostUpdateEvent(engine, delta, this));
-       this.emit('postupdate', new PostUpdateEvent(engine, delta, this));
-    }
-    /**
-     * Called by the Engine, draws the actor to the screen
-     * @param ctx   The rendering context
-     * @param delta The time since the last draw in milliseconds
-     */
-    public draw(ctx: CanvasRenderingContext2D, delta: number) {
-       ctx.save();
-       ctx.translate(this.pos.x, this.pos.y);       
-       ctx.scale(this.scale.x, this.scale.y);
-       ctx.rotate(this.rotation);
-
-       // translate canvas by anchor offset
-       ctx.save();
-       ctx.translate(-(this._width * this.anchor.x), -(this._height * this.anchor.y));
-
-       this.emit('predraw', new PreDrawEvent(ctx, delta, this));
-                    
-       if (this.currentDrawing) {
-          var drawing = this.currentDrawing;
-          // See https://github.com/excaliburjs/Excalibur/pull/619 for discussion on this formula          
-          var offsetX = (this._width - drawing.naturalWidth * drawing.scale.x) * this.anchor.x;
-          var offsetY = (this._height - drawing.naturalHeight * drawing.scale.y) * this.anchor.y;
-
-          if (this._effectsDirty) {
+   }
+
+   /**
+    * Perform euler integration at the specified time step
+    */
+   public integrate(delta: number) {
+      // Update placements based on linear algebra
+      var seconds = delta / 1000;
+
+      var totalAcc = this.acc.clone();
+      // Only active vanilla actors are affected by global acceleration
+      if (this.collisionType === CollisionType.Active) {
```

Comment:
Also fair point, if UIActor, Label, or Trigger are active they did do it on purpose. So probably okay for now.

My previous point is definitely out of scope for this issue and is a major change that needs more discussion ;)